tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1173,19 +1209,23 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     // dimensions; i.e. its most minor physical dimensions align with most minor
     // physical dimensions of the gemm output.
     absl::Span<const int64_t> broadcast_dims = broadcast->dimensions();
-    for (size_t i = 0; i < num_col_dims; ++i) {
-      int64_t dim = gemm->shape().layout().minor_to_major(i);
+    if (bitcast != nullptr) {
+      broadcast_dims = gemm->shape().dimensions();
```

Comment:
Why not check the bias vector is broadcast in the most major dimensions if there is a bitcast?

Also why set `broadcast_dims` in this branch? It is only used in the `else` branch.

Reply:
Resolved.